### PR TITLE
Add tryMappedPagedFile to PageCache and remove EXCLUSIVE map option

### PR DESCRIPTION
**Pull request by burqen**
This will act as a convenient utility when redirecting store copy to use PageCache directly for store files instead of using the file system.

### CODE REVIEW

File: community/io/src/main/java/org/neo4j/io/pagecache/PageCache.java [hunk 1]
```diff
@@ -61,6 +59,24 @@
      */
```
Comment (MishaDemianenko) [inline community/io/src/main/java/org/neo4j/io/pagecache/PageCache.java:58]: Previous line in java doc mention exclusive mapping conflicts - those are gone with this PR, so the reason for IOException should be updated as well

File: community/io/src/main/java/org/neo4j/io/pagecache/PageCache.java [hunk 2]
```diff
@@ -61,6 +59,24 @@
...
+     * false.
+     * <p>
+     * NOTE! User is responsible for closing the returned paged file.
+     *
+     * @param file The file to try to get the mapped paged file for.
+     * @return {@link Optional} containing the {@link PagedFile} mapped by this {@link PageCache} for given file, or an
+     * empty {@link Optional} if no mapping exist.
+     * @throws IOException if page cache has been closed or page eviction problems occur.
+     */
+    Optional<PagedFile> tryMappedPagedFile( File file ) throws IOException;
```
Comment (MishaDemianenko) [inline community/io/src/main/java/org/neo4j/io/pagecache/PageCache.java]: What do you think about renaming it to `findMappedFile` or `getMappedFile` instead?
With `try` in the begging it gives filling that we will try to map provided file (javadoc give explanation - yes, but we need to strive to make our code self-explanatory)

File: community/server/src/test/java/org/neo4j/server/rest/domain/RelationshipExpanderBuilderTest.java [hunk 3]
```diff
@@ -29,11 +29,10 @@
 import org.neo4j.graphdb.Relationship;
 import org.neo4j.graphdb.Transaction;
 import org.neo4j.kernel.impl.MyRelTypes;
-import org.neo4j.test.DatabaseRule;
-import org.neo4j.test.ImpermanentDatabaseRule;
+import org.neo4j.test.rule.DatabaseRule;
```
Comment (MishaDemianenko) [inline community/server/src/test/java/org/neo4j/server/rest/domain/RelationshipExpanderBuilderTest.java:32]: hm...why do you need to change this things, how do they work in master then?

**Comment by MishaDemianenko** [comment]
@burqen review round completed, 2 minor things and one small question

File: community/io/src/main/java/org/neo4j/io/pagecache/PageCache.java (hunk 2 above)
Comment (chrisvest) [inline community/io/src/main/java/org/neo4j/io/pagecache/PageCache.java]: I suggested the `try` prefix to hint at the optional return type. Another option could be `getExistingMapping`.

Comment (MishaDemianenko) [inline community/io/src/main/java/org/neo4j/io/pagecache/PageCache.java]: I like `getExistingMapping`! Can we please use it instead?

Comment (burqen) [inline community/io/src/main/java/org/neo4j/io/pagecache/PageCache.java]: 👍

File: community/server/src/test/java/org/neo4j/server/rest/domain/RelationshipExpanderBuilderTest.java (hunk 3 above)
Comment (burqen) [inline community/server/src/test/java/org/neo4j/server/rest/domain/RelationshipExpanderBuilderTest.java:32]: Have no idea how it compiles in master, but that's where they live
https://github.com/neo4j/neo4j/blob/3.1/community/kernel/src/test/java/org/neo4j/test/rule/DatabaseRule.java#L20

**Comment by burqen** [comment]
@MishaDemianenko Updated

**Pull request merged**